core/src/main/scala/org/broadinstitute/dsde/rawls/entities/compact/CompactEntityProvider.scala

```diff
@@ -116,9 +119,53 @@ class CompactEntityProvider(requestArguments: EntityRequestArguments, repository
 
   override def deleteEntities(entityRefs: Seq[AttributeEntityReference],
                               parentContext: RawlsRequestContext
-  ): Future[Int] = ???
+  ): Future[Int] =
+    repository.dataSource.inTransaction { _ =>
+      for {
+        entities: Seq[Option[CompactEntityRecord]] <- DBIO.sequence(
+          entityRefs.map(ref => repository.queries.getEntity(workspaceId, ref.entityType, ref.entityName))
+        )
+        entityIds = entities.flatten.map(_.id).toSet
+        // check if any of these entities are referenced by someone else
+        referencingEntities: Seq[AttributeEntityReference] <- repository.queries.getReferencingEntities(entityIds)
+        _ = if (referencingEntities.toSet.diff(entityRefs.toSet).size != 0) {
+          throw new DeleteEntitiesConflictException(referencingEntities.toSet)
+        }
+        // remove all references from these entities
+        _ <- repository.queries.deleteAllReferences(entityIds)
+        res <- repository.queries.batchHide(workspaceId, entityRefs)
+      } yield res.sum
+    }
 
-  override def deleteEntitiesOfType(entityType: String, parentContext: RawlsRequestContext): Future[Int] = ???
+  override def deleteEntitiesOfType(entityType: String, parentContext: RawlsRequestContext): Future[Int] =
+    repository.dataSource.inTransaction { _ =>
+      repository.queries.getEntitiesOfType(workspaceId, entityType).flatMap { entities =>
+        if (entities.isEmpty) {
+          DBIO.successful(0)
+        } else {
+          val entityIds = entities.map(_.id).toSet
+          for {
+            // check if any of these entities are referenced by someone else
+            referencingEntities: Seq[AttributeEntityReference] <- repository.queries.getReferencingEntities(entityIds)
+            referencingEntityTypes: Seq[String] = referencingEntities
+              .groupBy(_.entityType)
+              .keys
+              .filterNot(_ == entityType)
+              .toSeq
+            // Since we're removing all entities of entityType, we only need to check whether there are any other types included
+            _ = if (referencingEntityTypes.size > 0) {
+              throw new DeleteEntitiesOfTypeConflictException(referencingEntities.size)
+            }
+            // remove all references from these entities
+            _ <- repository.queries.deleteAllReferences(entityIds)
+            res <- repository.queries.batchHide(
+              workspaceId,
+              entities.map(entity => AttributeEntityReference(entity.entityType, entity.name))
+            )
```

Comment:
Because of the `batchHide` implementation, the resultant sql query here will use a `where … entity_type = ? and name in (?)` clause, where all it really needs is the much simpler `where … entity_type = ?` clause. It also requires passing a `Seq[AttributeEntityReference]` argument which could be potentially quite large - think of a data table with hundreds of thousands of rows. Could you optimize this?